src/application/polymarket/types.ts

```diff
@@ -80,31 +76,83 @@ export interface PolymarketUser {
   hasPolymarketAccount: boolean;
   safeWalletDetails?: SafeWallet;
   signIn: () => Promise<void>;
-  switchToPolygon: (provider: EIP1193Provider) => Promise<void>;
+  switchToPolygon: (wallet: Wallet) => Promise<void>;
 }
 
-interface SafeWallet {
+export interface SafeWallet {
   address: Hex;
   balance: number;
 }
 
-interface OrderDetails {
-  minimumTickSize: TickSize;
-  negRisk: boolean;
-  tokenId: string;
-  amount: number;
-}
-
 export interface PlaceOrderParameters {
   wallet: Wallet;
   funderAddress: Hex;
-  orderDetails: OrderDetails;
+  orderDetails: {
+    minimumTickSize: TickSize;
+    negRisk: boolean;
+    tokenId: string;
+    amount: number;
+  };
 }
 
+export type SaltedOrderData = OrderData & { salt: string };
+
 export interface OrderPlacer {
   reset: () => void;
-  place: (parameters: PlaceOrderParameters) => Promise<void>;
+  place: (properties: PlaceOrderParameters) => Promise<void>;
   isError: boolean;
   isPlaced: boolean;
   isPlacing: boolean;
 }
+
+export type TickSize = '0.1' | '0.01' | '0.001' | '0.0001';
+
+export interface Order {
+  order: {
+    salt: number;
+    maker: string;
+    signer: string;
+    taker: string;
+    tokenId: string;
+    makerAmount: string;
+    takerAmount: string;
+    side: 'BUY';
+  };
+  owner: string;
+  orderType: 'FOK';
+}
+
+export interface OrderData {
+  maker: string;
+  taker: string;
+  tokenId: string;
+  makerAmount: string;
+  takerAmount: string;
+  side: 0;
```

Comment:
Are the values like 'BUY', 'FOK', '0', and '2' static, non-changing values?

Reply:
it always has to be like that for our use-case, in general polymarket accepts some different values but we only care about these when making an order

Reply:
okay